packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/PsiMdbTreeUtil.kt

```diff
@@ -369,3 +375,17 @@ fun <T> PsiElement.findAllChildrenOfType(type: Class<T>): List<T> {
 
     return allChildren
 }
+
+/**
+ * Resolves to the first meaningful expression in a tree. Not all expressions have an important
+ * meaning for us (like for example, parenthesized expressions) so we drop them and get any
+ * inner expression that is relevant for us.
+ */
+fun PsiElement.meaningfulExpression(): PsiElement {
+    return when (this) {
+        // the children are: '(', YOUR_EXPR, ')'
+        // so we need the expression inside (index 1)
+        is PsiParenthesizedExpression -> children[1].meaningfulExpression()
```

Comment:
Shouldn't we assume that this index access might result in a null value or an index out of bounds exception?

Reply:
I believe this situation might come when the parenthesized expression is empty or incomplete?

Reply:
Ah true, good point, we can either return children[1] or this. Making a quick fix.